src/main/java/net/dries007/tfc/objects/entity/animal/EntityAnimalTFC.java

```diff
@@ -0,0 +1,352 @@
+                if (familiarity < 0.3f && (this.lastFed == -1 || this.lastFed - 1 < CalendarTFC.INSTANCE.getTotalDays()))
+                {
+                    familiarity -= 0.02 * (CalendarTFC.INSTANCE.getTotalDays() - this.lastFDecay);
+                    this.lastFDecay = CalendarTFC.INSTANCE.getTotalDays();
+                    if (familiarity < 0) familiarity = 0f;
+                    this.setFamiliarity(familiarity);
+                }
+            }
+            if (this.getGender() == Gender.MALE && this.isReadyToMate())
+            {
+                this.matingTime = CalendarTFC.INSTANCE.getCalendarTime();
+                if (findFemaleMate())
+                {
+                    this.setInLove(null);
+                }
+            }
+            if (this.lastFDecay > CalendarTFC.INSTANCE.getTotalDays())
+            {
+                //Calendar went backwards by command! this need to update
+                this.lastFDecay = CalendarTFC.INSTANCE.getTotalDays();
+            }
+            if (this.lastFed > -1 && this.lastFed > CalendarTFC.INSTANCE.getTotalDays())
+            {
+                //Calendar went backwards by command! this need to update
+                this.lastFed = CalendarTFC.INSTANCE.getTotalDays();
+            }
+            if (this.matingTime > -1 && this.matingTime > CalendarTFC.INSTANCE.getCalendarTime())
+            {
+                //Calendar went backwards by command! this need to update
+                this.matingTime = CalendarTFC.INSTANCE.getCalendarTime();
+            }
+            if (this.getBirthDay() > CalendarTFC.INSTANCE.getTotalDays())
+            {
+                //Calendar went backwards by command! this need to update
+                this.setBirthDay((int) CalendarTFC.INSTANCE.getTotalDays());
+            }
+        }
+    }
+
+    @Override
+    public void writeEntityToNBT(NBTTagCompound nbt)
+    {
+        super.writeEntityToNBT(nbt);
+        nbt.setBoolean("gender", getGender().toBool());
+        nbt.setInteger("birth", getBirthDay());
+        nbt.setLong("fed", lastFed);
+        nbt.setLong("decay", lastFDecay);
+        nbt.setBoolean("fertilized", this.fertilized);
+        nbt.setLong("mating", matingTime);
+        nbt.setFloat("familiarity", getFamiliarity());
+    }
+
+    @Override
+    public void readEntityFromNBT(NBTTagCompound nbt)
+    {
+        super.readEntityFromNBT(nbt);
+        this.setGender(Gender.fromBool(nbt.getBoolean("gender")));
+        this.setBirthDay(nbt.getInteger("birth"));
+        this.lastFed = nbt.getLong("fed");
+        this.lastFDecay = nbt.getLong("decay");
+        this.matingTime = nbt.getLong("mating");
+        this.fertilized = nbt.getBoolean("fertilized");
+        this.setFamiliarity(nbt.getFloat("familiarity"));
+
+    }
+
+    @Override
+    public boolean isBreedingItem(ItemStack stack)
+    {
+        return DEFAULT_BREEDING_ITEMS.contains(stack.getItem());
```

Comment:
```suggestion
        return OreDictionaryHelper.doesStackMatchOre(stack, "grain")
```
Other things that need to change:
 - since `grain` is already used as an ore dict name for all grain category items, that should probably be changed to be more specific, i.e. `foodCategoryGrain`. Then the ore dict name for grain can be added. However, proper ore dict names for food are already a part of #158, so for now just leave it as is, and it can be fixed once that is merged.